sdk/core/src/main/java/com/klaviyo/coresdk/model/UserInfo.kt

```diff
@@ -46,31 +46,42 @@ internal object UserInfo {
         get() = field.ifEmpty {
             // Attempts to read a UUID from the shared preferences.
             // If not found, generate a fresh one and save that to the data store
-            field = (Registry.dataStore.fetch(ANONYMOUS_ID.name) ?: "").ifEmpty {
+            field = (Registry.dataStore.fetch(ANONYMOUS.name) ?: "").ifEmpty {
                 val uuid = UUID.randomUUID().toString()
-                Registry.dataStore.store(ANONYMOUS_ID.name, uuid)
+                Registry.dataStore.store(ANONYMOUS.name, uuid)
                 uuid
             }
             return field
         }
 
-    fun reset() {
+    /**
+     * Updates [UserInfo] identifiers in state from a given [Profile] object
+     *
+     * @param profile
+     */
+    fun updateFromProfile(profile: Profile) = apply {
+        externalId = profile.identifier ?: externalId
```

Comment:
i think we should rename `identifier` to `externalId`

Reply:
thought i did! I missed one, thanks.

Reply:
Oh i see its in the profile class's attribute yeah